test/Orthogonal.hs

```diff
@@ -40,17 +40,17 @@ testSuite = testGroup "Orthogonal"
     ]
   ]
 
-prop_jacobi_de :: Rational -> Rational -> Property
-prop_jacobi_de a b = foldl' (.&&.) (property True) $
+prop_jacobi_de :: NonNegative Rational -> NonNegative Rational -> Property
+prop_jacobi_de (NonNegative a) (NonNegative b) = foldl' (.&&.) (property True) $
   zipWith (((=== 0) .) . de) [0..limit] (jacobi a b)
   where
     de :: Rational -> VPoly Rational -> VPoly Rational
     de n y = [1, 0, -1] * deriv (deriv y)
            + [b - a, - (a + b + 2)] * deriv y
            + [n * (n + a + b + 1)] * y
 
-prop_gegenbauer_de :: Rational -> Property
-prop_gegenbauer_de g = foldl' (.&&.) (property True) $
+prop_gegenbauer_de :: NonNegative Rational -> Property
+prop_gegenbauer_de (NonNegative g) = foldl' (.&&.) (property True) $
```

Comment:
Same here: when g is half-integer, one can compute a corresponding Gegenbauer polynomial directly.